supreme/src/jvmMain/kotlin/at/asitplus/signum/supreme/sign/PrivateKeySignerImpl.kt

```diff
@@ -0,0 +1,37 @@
+package at.asitplus.signum.supreme.sign
+
+import at.asitplus.catching
+import at.asitplus.signum.indispensable.*
+import at.asitplus.signum.supreme.dsl.DSL
+import at.asitplus.signum.supreme.dsl.DSLConfigureFn
+import at.asitplus.signum.supreme.os.JKSSignerConfiguration
+
+
+actual fun makePrivateKeySigner(
+    key: CryptoPrivateKey.RSA,
+    algorithm: SignatureAlgorithm.RSA
+): Signer.RSA = EphemeralSigner.RSA(config = EphemeralSignerConfiguration(), privateKey = key.toJcaPrivateKey().getOrThrow(), publicKey = key.publicKey, signatureAlgorithm = algorithm)
+
+actual fun makePrivateKeySigner(
+    key: CryptoPrivateKey.EC,
+    algorithm: SignatureAlgorithm.ECDSA
+): Signer.ECDSA = EphemeralSigner.EC(config = EphemeralSignerConfiguration(), privateKey = key.toJcaPrivateKey().getOrThrow(), publicKey = key.publicKey!!, signatureAlgorithm = algorithm)
+
+/**
+ * Creates a signer for the specified [privateKey]. Fails if the key type does not match the signature algorithm type (EC/RSA) or if it has no public key attached
+ * This JVM-specific variant allows for optionally specifying a provider
+ *
+ * @see JvmEphemeralSignerCompatibleConfiguration
+ *
+ */
+fun SignatureAlgorithm.signerFor(privateKey: CryptoPrivateKey<*>, configure: DSLConfigureFn<JvmEphemeralSignerCompatibleConfiguration>) = catching {
+    require(
+        (this is SignatureAlgorithm.ECDSA && privateKey is CryptoPrivateKey.EC) ||
+                (this is SignatureAlgorithm.RSA && privateKey is CryptoPrivateKey.RSA)
+    ){"Algorithm and Key mismatch: ${this::class.simpleName} + ${privateKey::class.simpleName}"}
+    when(this) {
+        is SignatureAlgorithm.ECDSA ->   EphemeralSigner.EC(config =DSL.resolve(::EphemeralSignerConfiguration, configure), privateKey = privateKey.toJcaPrivateKey().getOrThrow(), publicKey = privateKey.publicKey!! as CryptoPublicKey.EC, signatureAlgorithm = this)
```

Comment:
reformat into several lines to keep the column limit and increase readability
also apply auto-formatting to the whole file